### PR TITLE
Add CLI args for debug, nodeVersion, and npmVersion

### LINKED ISSUE #78: Add --debug option
Right now the cli suppresses some errors, but it would be nice to have a --debug option we can have people run that will not only show every error, but log some extra information as well.

### LINKED ISSUE #137: Meteor 1.4 deploy with modulus
Hello

version of node.js at the build is  "v0.10.41 (npm v3.9.6)"

Meteor 1.4 needs node.js 4.4.7

is it a way to fix it with modulus.

Best Regards

### PR DESCRIPTION
**Pull request by harlanj**
Support npmVersion, nodeVersion, and debug flags being sent on deploy

Closes #137
Closes #78 
References #106

### CODE REVIEW

**Comment by jackboberg** [comment]
👍

File: lib/commands/project.js [hunk 1]
```diff
@@ -668,10 +673,17 @@ project.deploy = function (projectName, dir, includeModules, registry, withTests
...
     function deployProject (selectedProject, fn) {
-      if (!selectedProject) {
-        return fn('You must deploy to a project.');
-      }
-      projectController.deploy(selectedProject.id, dir, includeModules, registry, withTests, function (err, domain) {
+      if (!selectedProject) return fn('You must deploy to a project.');
+
+      options = {
+        registry: registry,
+        withTests: !!withTests,
```
Comment (tzmanics) [inline lib/commands/project.js]: Just out of curiosity, why the `!!`

Comment (harlanj) [inline lib/commands/project.js]: @tzmanics It's noted [here](https://github.com/onmodulus/modulus-cli/pull/139/files#diff-e5001988c284bf49302c6986b4b88a3bL312). I just moved the checking to a different file. Also it's to make it a boolean if it already isn't one.

**Comment by franvarney** [comment]
👍

**Comment by tzmanics** [comment]
Should we update the `README` for these changes in this PR?

**Comment by harlanj** [comment]
Yea. Making those changes now

**Comment by harlanj** [comment]
@tzmanics changes made

File: lib/routes/project.js [hunk 2]
```diff
@@ -164,6 +164,9 @@ module.exports = function (modulus) {
     this.line('    -m, --include-modules   Flag that indicates whether or not to include the node_modules folder in the bundle.'.input);
     this.line('    -r, --registry          Specify a registry to install modules from.'.input);
     this.line('    -w, --with-tests        Specifies that tests should be run before deploying this project.'.input);
+    this.line('        --nodeVersion       Specified version of node to use when running application.'.input);
```
Comment (harlanj) [inline lib/routes/project.js]: Those are typos. I just corrected them to be `--node-version` and `--npm-version`

Comment (harlanj) [inline lib/routes/project.js]: @tzmanics ^

**Comment by tzmanics** [comment]
👍

**Pull request merged**